api/codelists.go

```diff
@@ -15,22 +15,38 @@ import (
 func (c *CodeListAPI) getCodeLists(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	filterBy := r.URL.Query().Get("type")
+	logData := log.Data{}
+	offsetParameter := r.URL.Query().Get("offset")
+	limitParameter := r.URL.Query().Get("limit")
+	offset := c.defaultOffset
+	limit := c.defaultLimit
+	var err error
 
-	// get limit from query parameters, or default value
-	limit, err := GetPositiveIntQueryParameter(r.URL.Query(), "limit", c.defaultLimit)
-	if err != nil {
-		handleError(ctx, "failed to obtain a positive integer value for limit query parameter", log.Data{"limit": limit}, err, w)
-		return
+	log.Event(ctx, "attempting to get code lists", log.INFO, log.Data{"type": filterBy})
+
+	if offsetParameter != "" {
+		logData["offset"] = offsetParameter
+		offset, err = ValidatePositiveInt(offsetParameter)
+		if err != nil {
+			handleError(ctx, "failed to obtain a positive integer value for offset query parameter", log.Data{}, err, w)
+			return
+		}
 	}
 
-	// get offset from query parameters, or default value
-	offset, err := GetPositiveIntQueryParameter(r.URL.Query(), "offset", c.defaultOffset)
-	if err != nil {
-		handleError(ctx, "failed to obtain a positive integer value for offset query parameter", log.Data{"offset": offset}, err, w)
-		return
+	if limitParameter != "" {
+		logData["limit"] = limitParameter
+		limit, err = ValidatePositiveInt(limitParameter)
+		if err != nil {
+			handleError(ctx, "failed to obtain a positive integer value for offset query parameter", log.Data{}, err, w)
+			return
+		}
 	}
 
-	log.Event(ctx, "attempting to get code lists", log.INFO, log.Data{"type": filterBy})
+	if limit > c.maxLimit {
+		logData["max_limit"] = c.maxLimit
+		handleError(ctx, "limit is greater than the maximum allowed", log.Data{}, err, w)
```

Comment:
Not sure what this err would register as? Should it be something like err = errors.something?

Reply:
I believe this is the correct implementation. `err` is defined on line 23 as type error, then passed to the `handleError` function which returns an error based on the following:
```
if err == driver.ErrNotFound {
		http.Error(w, err.Error(), http.StatusNotFound)
	} else {
		http.Error(w, internalServerErr, http.StatusInternalServerError)
	}
```
So I don't think there is any need for an if statement here